### ISSUE TITLE
Temporarily Remove Accounts Returning Empty Responses

### ISSUE THREAD

**Issue by PhoenixICE** [favll/pogom#183]
Feature Request:

If an account hits the "empty response issue" - can you take it out of the pool of accounts to scan for like x seconds (I would say like 30 seconds or something)

**Comment by PhoenixICE** [comment]
either that or allow us to have separate pool of accounts that can swap in with accounts that get the empty response

Pool A - Active Scanning Accounts
Pool B - InActive Scanning Accounts
Pool C - OnHold Accounts

Move Pool A accounts into Pool C when hitting empty, and move a Pool B Account into Pool A to make up the difference

**Comment by Lionir** [comment]
Seems like a great idea to me.

**Comment by nborrmann** [comment]
The problem is, it's kinda hard to identify whether or not an account is banned. It does return a perfectly valid response, it's just nothing in there.
I tested it in a rural area and there it's quite common that there is no stuff in the scanned area.

**Comment by johved** [comment]
Yeah thats a problem @nborrmann. Is it possible to get the distance to nearest pokestop? because i guess that would be a non-rural definer. For example if an account gets no returns for 3mins in close distance to pokestops it is most likely softbanned. To make it even better you could perhaps gather data from the google maps API about whether it is on land or not

**Comment by djv2** [comment]
great idea, +1!!

@nborrmann you already implented the code to see if its softbanned
" if sum(len(cell.keys()) for cell in cells) == len(cells) * 2:" yes?

**Comment by nborrmann** [comment]
@djv2 That checks for an empty response. But in rural areas that might be perfectly fine, because there's just nothing there.

**Comment by djv2** [comment]
@nborrmann the only thing i can think of is still develop the functionality with your code (check empty response) but as a flag. i 100% agree in rural that might be normal, but i know like at least 75%+ of us are scanning cities, let us the users decide whether to use that as a feature

**Comment by djv2** [comment]
@nborrmann to temp add this feature, can't i just edit parse_map(map_dict) and put everything under your
`if sum(len(cell.keys()) for cell in cells) == len(cells) * 2: log.warning("Received valid response but without any data. Possibly rate-limited?")`
in an else statement? so if it returns empty it won't use it. that should speed things up yes?

**Comment by alexanderhektor** [comment]
@nborrmann True (empty responses in rural areas).

A little bit more complex, but here's an idea to determine a probability of validness for the empty response:
- spawn points are always the same. (I'm not sure if they're exact lat/long every time), but afaik we haven't seen changes in spawn points yet.
- spawn points are also predictable in time (once every full hour (e.g. 08:33, 9:33, etc.). Duration is not exactly predictable afaik. (mostly 15m, but 30, 45 or even 1h(?) is possible).

So after a while, the data from the local db should be sufficient to know if there should actually be a pokemon in that scan, or not. (e.g. previous x scans have shown a spawn at every hour there, but now nothing has spawned?)

Those proven assumptions can also be used to for a more intelligent scanning pattern (e.g. not scanning areas without spawns (or only to check if the spawn points have changed from time to time), scanning spawns at the right times etc.)

**Comment by PhoenixICE** [comment]
@nborrmann acouple things we could try here:

Set the re-use interval to random interval from 15 - 20 seconds of accounts rather then 10 seconds

If an account returns an empty response x times then deem it temporarily banned

Do a global API rate limit - so the api can only be called X times per second/minute

**Comment by PhoenixICE** [comment]
If I could understand python I would help, but its so weird compared to C#

**Comment by wHaTsen** [comment]
Wouldn't it be possible to flag an account that returned 5 empty responses in a row and scan an area where we know that there is definitely a pokemon?

Account 1 scans an area where a pokemon was detected.
Account 2 scans an area without returns an empty response. After 5 scans there were only empty responses. So Account 2 scans the area where Account 1 found a pokemon. If Account 2 returns a pokemon it is not blocked. Otherwise it will be temporarily removed. 

I think I do not have to mention that the pokemon has to be there when Account 2 scans the area.

**Comment by djv2** [comment]
bump - wish this was a priority instead of devs adding features. scanning needs to be #1 :100:
